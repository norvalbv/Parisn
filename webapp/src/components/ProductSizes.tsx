import React, { useState } from 'react';
import Button from './Button';

interface ProductSizesProps {
  sizes: string[];
  classes?: string;
  onClick?: (arg: string) => void;
  selectedSize?: string;
}

const ProductSizes = ({ sizes, classes, onClick, selectedSize }: ProductSizesProps) => {
  const [selected, setSelected] = useState<null | string>(null);

  const handleClick = (size: string) => {
    if (onClick) return onClick(size);
    setSelected(size);
  };

  return (
    <div className={`flex gap-4 ${classes}`}>
      {sizes.map((size) => (
        <Button
          text={size}
          size="xs"
          width={30}
          borderRequired="bottom"
          hoverColorRequired={false}
          classes={`hover:border-b-pink-500 hover:scale-125 ${
            selectedSize === size || selected === size ? 'scale-125 border-b-pink-500' : ''
          }`}
          onClick={() => handleClick(size)}
        />
      ))}
    </div>
  );
};

export default React.memo(ProductSizes);
