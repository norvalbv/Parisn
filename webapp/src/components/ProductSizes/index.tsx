import React, { useState } from 'react';
import { Stock } from '../../types';
import Button from '../Button';

export interface ProductSizesProps {
  sizes: Stock;
  classes?: string;
  onClick?: (arg: string) => void;
  selectedSize?: string;
}

const ProductSizes = ({ sizes, classes, onClick, selectedSize }: ProductSizesProps) => {
  const [selected, setSelected] = useState<null | string>(null);

  const handleClick = (size: string) => {
    if (onClick) return onClick(size);
    setSelected(size);
  };

  const sortedList = Object.entries(sizes).sort((a, b) => a[1] - b[1]);

  return (
    <div className={`flex gap-4 ${classes}`}>
      {sortedList.map((size, i) => {
        const productSize =
          size[0].slice(0, 1).toLowerCase() === 'e' ? 'xl' : size[0].slice(0, 1).toLowerCase();
        const stockCount = Number(size[1]);
        return (
          <Button
            key={i}
            text={productSize}
            size="xs"
            width="1.5rem"
            borderRequired="bottom"
            hoverColorRequired={false}
            disabled={Boolean(!size[1])}
            classes={`${size[1] ? 'hover:border-b-pink-500 hover:scale-125' : ''} ${
              (selectedSize?.toLowerCase() === productSize && stockCount > 0) ||
              (selected === productSize && stockCount > 0)
                ? 'scale-125 border-b-pink-500'
                : ''
            }`}
            onClick={() => handleClick(productSize)}
          />
        );
      })}
    </div>
  );
};

export default React.memo(ProductSizes);
