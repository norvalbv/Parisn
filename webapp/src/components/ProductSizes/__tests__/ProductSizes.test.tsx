import React, { FC } from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

import ProductSizes, { ProductSizesProps } from '..';

const ProductSizesWithRequiredProps: FC<Required<ProductSizesProps>> = ProductSizes;

describe('<ProductSizes />', () => {
  test('ProductSizes props with mandatory props', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <ProductSizes sizes={{ Small: 5, Medium: 4, Large: 3, ExtraLarge: 2 }} />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('ProductSizes props with all props as required', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <ProductSizesWithRequiredProps
            sizes={{ Small: 5, Medium: 4, Large: 3, ExtraLarge: 2 }}
            classes="bg-red-500"
            onClick={jest.fn()}
            selectedSize="Medium"
          />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
