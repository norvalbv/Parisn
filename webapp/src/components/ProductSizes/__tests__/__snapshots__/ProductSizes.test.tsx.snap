// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductSizes /> ProductSizes props with all props as required 1`] = `
<div
  className="flex gap-4 bg-red-500"
>
  <button
    className="relative inline-flex items-center justify-center false uppercase  px-1 py-2 text-sm hover:border-b-pink-500 hover:scale-125  border-b py-4 "
    disabled={false}
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    role="button"
    style={
      {
        "backgroundColor": undefined,
        "color": undefined,
        "width": "1.5rem",
      }
    }
    type="button"
  >
    <span
      className="inline-block font-semibold"
    >
      s
    </span>
  </button>
  <button
    className="relative inline-flex items-center justify-center false uppercase  px-1 py-2 text-sm hover:border-b-pink-500 hover:scale-125  border-b py-4 "
    disabled={false}
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    role="button"
    style={
      {
        "backgroundColor": undefined,
        "color": undefined,
        "width": "1.5rem",
      }
    }
    type="button"
  >
    <span
      className="inline-block font-semibold"
    >
      m
    </span>
  </button>
  <button
    className="relative inline-flex items-center justify-center false uppercase  px-1 py-2 text-sm hover:border-b-pink-500 hover:scale-125  border-b py-4 "
    disabled={false}
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    role="button"
    style={
      {
        "backgroundColor": undefined,
        "color": undefined,
        "width": "1.5rem",
      }
    }
    type="button"
  >
    <span
      className="inline-block font-semibold"
    >
      l
    </span>
  </button>
  <button
    className="relative inline-flex items-center justify-center false uppercase  px-1 py-2 text-sm hover:border-b-pink-500 hover:scale-125  border-b py-4 "
    disabled={false}
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    role="button"
    style={
      {
        "backgroundColor": undefined,
        "color": undefined,
        "width": "1.5rem",
      }
    }
    type="button"
  >
    <span
      className="inline-block font-semibold"
    >
      xl
    </span>
  </button>
</div>
`;

exports[`<ProductSizes /> ProductSizes props with mandatory props 1`] = `
<div
  className="flex gap-4 undefined"
>
  <button
    className="relative inline-flex items-center justify-center false uppercase  px-1 py-2 text-sm hover:border-b-pink-500 hover:scale-125  border-b py-4 "
    disabled={false}
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    role="button"
    style={
      {
        "backgroundColor": undefined,
        "color": undefined,
        "width": "1.5rem",
      }
    }
    type="button"
  >
    <span
      className="inline-block font-semibold"
    >
      s
    </span>
  </button>
  <button
    className="relative inline-flex items-center justify-center false uppercase  px-1 py-2 text-sm hover:border-b-pink-500 hover:scale-125  border-b py-4 "
    disabled={false}
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    role="button"
    style={
      {
        "backgroundColor": undefined,
        "color": undefined,
        "width": "1.5rem",
      }
    }
    type="button"
  >
    <span
      className="inline-block font-semibold"
    >
      m
    </span>
  </button>
  <button
    className="relative inline-flex items-center justify-center false uppercase  px-1 py-2 text-sm hover:border-b-pink-500 hover:scale-125  border-b py-4 "
    disabled={false}
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    role="button"
    style={
      {
        "backgroundColor": undefined,
        "color": undefined,
        "width": "1.5rem",
      }
    }
    type="button"
  >
    <span
      className="inline-block font-semibold"
    >
      l
    </span>
  </button>
  <button
    className="relative inline-flex items-center justify-center false uppercase  px-1 py-2 text-sm hover:border-b-pink-500 hover:scale-125  border-b py-4 "
    disabled={false}
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    role="button"
    style={
      {
        "backgroundColor": undefined,
        "color": undefined,
        "width": "1.5rem",
      }
    }
    type="button"
  >
    <span
      className="inline-block font-semibold"
    >
      xl
    </span>
  </button>
</div>
`;
